Resolve an issue with "=" delimeter, change previous namings

alexsaber opened the pull request:
## Basics

Do not describe the purpose here but:

- [ ] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [ ] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [ ] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)

## Review

Remove the line below and add the "work in progress" label if you do
not want the PR to be reviewed:

@markus2330 please review my pull request

markus2330 commented:
Thank you for the further fix! Please make sure to read the output of the build server. Release notes are missing.

markus2330 commented:
And please also mention https://build.libelektra.org/jenkins/job/libelektra/job/master/lastSuccessfulBuild/artifact/artifacts/debian-stable-mingw-w64/ or didn't these binaries work?

alexsaber commented:
Thank you for you feedbak. Yes, I see, some tests are failing. For example, "key = something" expected, but "key=something" received.
Where are these tests defined? I am sorry, I have almost no experience with C/C++... If you point me where the tests are, I can fix them of course. Thanks.

markus2330 commented:
Most of the test cases are in src/plugins/ini/testmod_ini.c and tests/shell, one test case seem to be in src/plugins/blockresolver/testmod_blockresolver.c

alexsaber commented:
@markus2330 Thank you very much for helping me out. I could find a problem in the testmod_blockresolver and fix it. However, I still do not understand why these tests are failing. 
         35 - testshell_markdown_base64 (Failed)
	 36 - testmod_blockresolver (Failed)
	 37 - testshell_markdown_blockresolver (Failed)
	 46 - testshell_markdown_conditionals (Failed)
	 72 - testmod_ini (Failed)
	 73 - testshell_markdown_ini (Failed)
	 91 - testshell_markdown_mozprefs (Failed)
	166 - testshell_markdown_msr_syntax (Failed)
Where I can find why are these tests failing exactly?

sanssecours commented on src/plugins/ini/README.md:
```diff
@@ -1,18 +1,18 @@
-- infos = Information about the ini plugin is in keys below
-- infos/author = Thomas Waser <thomas.waser@libelektra.org>
-- infos/licence = BSD
+- infos=Information about the ini plugin is in keys below
```
I do not think that it is a good idea to remove these spaces in the plugin contract, since the function [`generate_readme`](https://github.com/ElektraInitiative/libelektra/blob/master/cmake/Modules/LibAddMacros.cmake#L598-L754) seems to require them.

alexsaber commented:
@sanssecours Thanks for your comment. I changed it back.
@markus2330 I am sorry I bother you again. I see testmod_blockresolver causes problems again. 
This line fails
succeed_if (compare_line_files (srcdir_file (compareName), foutname), "files do not match as expected");
What does this check exactly do? If I try to trace it back, it still does not make sense to me why my changes broke this test. I could find neither documentation nor comments inside in code to get a better understanding of this use case. Can you please help me out?

kodebach commented:
First of all, please make sure to always [add a line to the release notes](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2541/4/pipeline#step-290-log-7) before creating a PR. Otherwise the the Jenkins builds don't run and you will just waste time. You can always change the line later, if you need to.

`srcdir_file` constructs the path to a file in the source directory of the plugin. And `elektraFilename()` returns the name of a temporary file. So the line question actually compares the temporary file created by the `kdbSet` call with the file `src/plugins/blockresolver/blockresolver/compare.block`. 

The [Travis log](https://travis-ci.org/ElektraInitiative/libelektra/jobs/513411729#L1666) indicates, that you probably changed the output of the plugin, so either revert that change or change the `compare.block` file to match the expected output.

---
PS. you can link to code snippets instead of copying the line:
https://help.github.com/en/articles/creating-a-permanent-link-to-a-code-snippet

kodebach commented:
There are still some tests failing, because of the whitespace arround `=`. You are also still missing the line in the release notes.

alexsaber commented:
@kodebach thanks for your comment! Do you mean the tests are failing because I changed the behavior it interprets the whitespaces around '=' or because somewhere in the test data there are still the whitespaces? If it is the second one, I cannot find anything written in the old format anymore. I change all the test data to the new format, meaning I removed the whitespaces. 
I will add the line in the release notes, thank you.

kodebach commented:
Lets take for example [this Travis job](https://travis-ci.org/ElektraInitiative/libelektra/jobs/513864204):

As you can see in [Line 1881](https://travis-ci.org/ElektraInitiative/libelektra/jobs/513864204#L1881) the `testshell_markdown_base64` test fails:
```
141/226 Test  #44: testshell_markdown_base64 ................***Failed    1.09 sec
```

The following lines show you the output. [Line 1929-1932](https://travis-ci.org/ElektraInitiative/libelektra/jobs/513864204#L1929-L1932) for example tell you one of the problems that occurred:

```
CMD: kdb file user/tests/base64 | xargs cat
RET: 0
STDOUT: binary="@BASE64AGLuy2N/AAA="
=== FAILED stdout does not match expected pattern binary = "@BASE64[a-zA-Z0-9+/]+={0,2}"
```

The command `kdb file user/tests/base64 | xargs cat` was expected to produce output matching the regex `binary = "@BASE64[a-zA-Z0-9+/]+={0,2}"`, but instead produced `binary="@BASE64AGLuy2N/AAA="`. As you can see, the problem the test expected spaces arround the `=`, but there weren't any.

If you don't know, the `kdb file` command outputs the path of the file in which a key is stored. So the whole `kdb file user/tests/base64 | xargs cat` simply prints the file containing the key `user/tests/base64`.

alexsaber commented:
Thank you for you comment!
I do understand that the test expected to get `@BASE64[a-zA-Z0-9+/]+={0,2}`, but got `@BASE64AGLuy2N/AAA=` instead.
I tried to change `@BASE64[a-zA-Z0-9+/]+={0,2}` to `@BASE64[a-zA-Z0-9+/]*={0,2}`, but it didn't help. So how to make this test to not expect whitespaces around `=`? I changed the way the ini plugin works with the whitespaces around `=` and I tripple-checked if I missed something. In my opinion, it should work.

kodebach commented:
You read the regex wrong. The problem is not the `@BASE64[a-zA-Z0-9+/]+={0,2}` part. That is fine, and it matches `@BASE64AGLuy2N/AAA=`. The problem is that it also expects `binary = ` at the start, but it got `binary=`. So you need to change `binary = "@BASE64[a-zA-Z0-9+/]+={0,2}"` into `binary="@BASE64[a-zA-Z0-9+/]+={0,2}"`.

The problems with other tests are similar. Just look for an ` = ` (<space><equal sign><space>) and try replacing it by `=` (<equal sign>).

kodebach commented:
It seems there are some formatting problems... Please reformat the files in question, so that the build server can do its job

alexsaber commented:
Thank you for your comment!
Ok, I see that testscr_check_formatting is failing. You are mentioning reformatiing of some files. Which files do you mean? I haven't changed the files this test if using. I think so.

kodebach commented:
The file that fails the check is [`src/plugins/ini/README.md`](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2541/8/pipeline/223#step-310-log-58), but you can always just run all of the `reformat-*` scripts (or if you rebase the new `reformat-all` script). The reformatting script won't do any damage if everything is perfect already.

sanssecours commented:
> Which files do you mean? 

As you can see [here](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2541/8/pipeline/223), the files

- `doc/INSTALL.md` and
- `src/plugins/ini/README.md`

are not formatted correctly. You can use the description in the link above to fix this problem.

alexsaber commented:
I see. Thank you very much for your support!

alexsaber commented:
Ok, I still need to work on formatting...

alexsaber commented:
I will update my fork and create a new pull request.
